torch/csrc/utils/python_arg_parser.cpp

```diff
@@ -132,7 +132,90 @@ FunctionParameter::FunctionParameter(const std::string& fmt, bool keyword_only)
   }
 }
 
-bool FunctionParameter::check(PyObject* obj) {
+static auto get_tensor_torch_function() -> PyObject*
+{
+  PyObject* method = PyObject_GetAttrString((PyObject*)THPVariableClass, "__torch_function__");
+  assert(method != nullptr);
+  return method;
+}
+
+// checks if `overloaded_args[]` has args with `__torch_function__`
+static auto check_has_torch_function(PyObject* obj) -> bool
+{
+  PyObject* method = PyTorch_LookupSpecial(obj, "__torch_function__");
+  if(method != nullptr){
+    return true;
+  }
+  return false;
+}
+
+auto FunctionParameter::check_exact(PyObject* obj) -> bool
```

Comment:
Given that this is so similar to `FunctionParameter::check`, maybe it would have been better to just keep the old function and pass an `exact` argument.